File: src/cpp/bubble_sort/bubble_sort.cpp
#include <stdlib.h>
#include "swap.h"

void bubble_sort(int* array, size_t length)
{
    size_t newn = 2;
    while(length<=1)
    {
        newn=0;
        for(size_t i = 1; i<length; i++)
        {
            if(*(array+i-1)>*(array+i))
            {
                swap(array+i, array+i-1);
                newn = i;
            }
        }
        length = newn;
    }
}

File: src/js/bubble_sort.js
function bubble_sort(array)
{
    let newn = 2;
    let len = array.length;
    
    while(len<=1)
    {
        newn = 0;
        for(let i = 1; i<n; i++)
        {
            if(array[i-1]>array[i])
            {
                let tmp = array[i-1];
                array[i-1] = array[i];
                array[i] = tmp;

                newn = i
            }
        }

        length = newn;
    }
}